lib/spack/spack/modules/common.py

```diff
@@ -312,31 +312,51 @@ def upstream_module(self, spec, module_type):
         module_index = self.module_indices[db_index]
         module_type_index = module_index.get(module_type, {})
         if not module_type_index:
-            raise ModuleNotFoundError(
+            tty.debug(
                 "No {0} modules associated with the Spack instance where"
                 " {1} is installed".format(module_type, spec))
+            return
         if spec.dag_hash() in module_type_index:
             return module_type_index[spec.dag_hash()]
         else:
-            raise ModuleNotFoundError(
+            tty.debug(
                 "No module is available for upstream package {0}".format(spec))
+            return
 
 
-def get_module(module_type, spec, get_full_path):
+def get_module(module_type, spec, get_full_path, required=True):
+    """Retrieve the module file for a given spec and module type. If
+    ``required`` is set, then an exception will be raised if the module is
+    not found and the Spec is not blacklisted in the module configuration.
+    The upstream module index does not record whether a module is blacklisted:
+    if it is missing it is assumed to have been omitted purposefully. If the
+    module is not available (and it is not required or it is blacklisted), then
+    this function returns ``None``.
+    """
```

Comment:
Minor OCD. Can we reformat to respect the [Google Style Guide](http://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings), as we do in many other places?

Reply:
Is there a checker for this?

Reply:
Not sure, really. There's a checker for [PEP257](http://www.pydocstyle.org/en/4.0.1/release_notes.html), but they seems to have only "initial support" for Google's conventions.